### ISSUE TITLE
Purpose of NanoHttpd.QUERY_STRING

### ISSUE THREAD

**Issue by stefan-reich** [NanoHttpd/nanohttpd#534]
Hi,

I am wondering why this parameter is passed to serve(). Can I just hack NanoHTTPD to not include this? Like, removing this line:

`        parms.put(NanoHTTPD.QUERY_STRING_PARAMETER, session.getQueryParameterString());`

The parameter is annoying when I want to put all the parameters into a link later (have to filter it out manually).

Thanks for a great product

**Comment by LordFokas** [comment]
Yeah it shouldn't be there in the first place, but this library was built on top of hacks and half-assed solutions the fact that despite all that it works as well as it does is remarkable. That's another thing that I aimed to fix in v3.
